shared/patches/recyclerlistview+4.2.0.patch

```diff
@@ -0,0 +1,44 @@
+diff --git a/node_modules/recyclerlistview/dist/reactnative/core/layoutmanager/LayoutManager.js b/node_modules/recyclerlistview/dist/reactnative/core/layoutmanager/LayoutManager.js
+index 9cd3c57..f65012e 100644
+--- a/node_modules/recyclerlistview/dist/reactnative/core/layoutmanager/LayoutManager.js
++++ b/node_modules/recyclerlistview/dist/reactnative/core/layoutmanager/LayoutManager.js
+@@ -114,7 +114,7 @@ var WrapGridLayoutManager = /** @class */ (function (_super) {
+         for (var i = startIndex; i < itemCount; i++) {
+             oldLayout = this._layouts[i];
+             var layoutType = this._layoutProvider.getLayoutTypeForIndex(i);
+-            if (oldLayout && oldLayout.isOverridden && oldLayout.type === layoutType) {
++            if (oldLayout && oldLayout.isOverridden /*&& oldLayout.type === layoutType*/) {
+                 itemDim.height = oldLayout.height;
+                 itemDim.width = oldLayout.width;
+             }
+@@ -145,9 +145,12 @@ var WrapGridLayoutManager = /** @class */ (function (_super) {
+                 itemRect.x = startX;
+                 itemRect.y = startY;
+                 itemRect.type = layoutType;
++		const nextOverriden = !!itemRect.isOverridden && itemRect.width === itemDim.width && itemRect.height === itemDim.height;
++		itemRect.isOverridden = nextOverriden
+                 itemRect.width = itemDim.width;
+                 itemRect.height = itemDim.height;
+             }
++
+             if (this._isHorizontal) {
+                 startY += itemDim.height;
+             }
+diff --git a/node_modules/recyclerlistview/dist/reactnative/platform/reactnative/viewrenderer/ViewRenderer.js b/node_modules/recyclerlistview/dist/reactnative/platform/reactnative/viewrenderer/ViewRenderer.js
+index d9c7031..dfc6d92 100644
+--- a/node_modules/recyclerlistview/dist/reactnative/platform/reactnative/viewrenderer/ViewRenderer.js
++++ b/node_modules/recyclerlistview/dist/reactnative/platform/reactnative/viewrenderer/ViewRenderer.js
+@@ -44,9 +44,10 @@ var ViewRenderer = /** @class */ (function (_super) {
+         };
+         _this._onLayout = function (event) {
+             //Preventing layout thrashing in super fast scrolls where RN messes up onLayout event
+-            var xDiff = Math.abs(_this.props.x - event.nativeEvent.layout.x);
```

Comment:
i haven't see this messed up RN `onLayout` event but, this can trip up the FlashList as it makes an assumption that when `reportItemLayout` is called the dimensions have actually been updated (https://github.com/Shopify/flash-list/blob/main/src/GridLayoutProviderWithProps.ts#L104). So we just disable this path